server/eclipse-project/src/main/java/us/freeandfair/corla/csv/DominionCVRExportParser.java

```diff
@@ -358,6 +386,51 @@ private CastVoteRecord extractCVR(final CSVRecord the_line) {
   }
   
   /**
+   * Processes the headers from the specified CSV record. This includes checking
+   * for the use of forbidden headers, and that all required headers are 
+   * present.
+   * 
+   * @return true if the headers are OK, false otherwise; this method also
+   * sets the error message if necessary.
+   */
+  private boolean processHeaders(final CSVRecord the_line) {
+    boolean result = true;
+    
+    // the explanations line includes the column names for the non-contest/choice
+    // columns, so let's get those
+    
+    for (int i = 0; i < my_first_contest_column; i++) {
+      my_columns.put(the_line.get(i), i);
+    }
+    
+    // let's make sure none of our prohibited headers are present
+    
+    for (final String h : PROHIBITED_HEADERS) {
+      if (my_columns.get(h) != null) {
+        result = false;
+        my_error_message = h + " column present in import file";
+        break;
+      }
+    }
+    
+    // let's make sure all of our required headers are present
+    
+    if (result) {
+      result &= my_columns.get(CVR_NUMBER_HEADER) != null;
+      result &= my_columns.get(TABULATOR_NUMBER_HEADER) != null;
+      result &= my_columns.get(BATCH_ID_HEADER) != null;
+      result &= my_columns.get(RECORD_ID_HEADER) != null;
+      result &= my_columns.get(IMPRINTED_ID_HEADER) != null;
+      result &= my_columns.get(BALLOT_TYPE_HEADER) != null;
+      if (!result) {
+        my_error_message = "missing required columns in import file";
```

Comment:
Could we build up a list of the missing column names, and include them in the error message?

Reply:
Yes. Yes we can.